include/Options/Abstract_Option.php

```diff
@@ -225,4 +226,58 @@ abstract protected function get_data_structure(): array;
 	 * @return string
 	 */
 	abstract protected function get_description(): string;
+
+	/**
+	 * Returns a list of language terms.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $fields Optional. Term fields to query for (see `WP_Term_Query` for possible values). Default is `'all'`.
+	 * @return array
+	 *
+	 * @phpstan-param 'all'|'all_with_object_id'|'names'|'slugs'|'id=>name'|'id=>slug'|'ids'|'tt_ids' $fields
+	 * @phpstan-return (
+	 *     $fields is 'all'|'all_with_object_id' ? list<WP_Term> : (
+	 *         $fields is 'names'|'slugs'|'id=>name'|'id=>slug' ? array<int, string> : array<int, int>
+	 *     )
+	 * )
+	 */
+	protected function get_language_terms( string $fields = 'all' ): array {
```

Comment:
The `$fields` parameter is overkill IMHO. It's only used once and can be filtered by the calling code easily.
In other words, it does things we don't need, adds complexity and maintenance ( `$fields` doc must be aligned with `get_terms()`).

Reply:
I disagree. If it is used (at least once), that means there is at least another case where it isn't used (in `Domains`), so there are 2 different results.
If we remove it we will add complexity instead of removing it. In `Nav_Menus` we do:

```php
$language_terms = $this->get_language_terms( 'slugs' );
// ...
$unknown_langs = array_diff_key( $unknown_langs, array_flip( $language_terms ) );
```

Without this parameter we will have to create an intermediate array containing the language slugs, which is exactly what this parameter does.

Reply:
As you said, it's use only once so the complexity is added into a new method (used once).
Also, you could simplify using `wp_list_pluck()` for instance.

Reply:
Done in [d995085](https://github.com/polylang/polylang/pull/1610/commits/d995085afda0b8a5df6f363f4412429b7d83df73).